File: src/java/foundgine-providers/src/main/java/com/foundgine/providers/storage/sql/query/SqlParameterBinding.java
package com.foundgine.providers.storage.sql.query;

import com.foundgine.core.semantic.planning.mutation.MutationValueReference;

/**
 * Binds one physical SQL parameter to a literal, mutation value, or execution
 * context path.
 */
public record SqlParameterBinding(String name, Object value, MutationValueReference source, String contextPath,
		Class<?> clrType) {
	public SqlParameterBinding(String name, Object value) {
		this(name, value, null, null, null);
	}

	public SqlParameterBinding(String name, Object value, MutationValueReference source) {
		this(name, value, source, null, null);
	}

	public SqlParameterBinding(String name, Object value, MutationValueReference source, String contextPath) {
		this(name, value, source, contextPath, null);
	}
}

File: src/csharp/Foundgine.Providers/Storage/Sql/Query/SqlParameterBinding.cs
using Foundgine.Core.Semantic.Planning.Mutation;

namespace Foundgine.Providers.Storage.Sql.Query;

public sealed record SqlParameterBinding(
    string Name,
    object? Value,
    MutationValueReference? Source = null,
    string? ContextPath = null,
    Type? ClrType = null);
